packages/salesforcedx-vscode-core/src/commands/util/sfdxCommandlet.ts

```diff
@@ -28,8 +28,23 @@ import { taskViewService } from '../../statuses';
 import { telemetryService } from '../../telemetry';
 import { getRootWorkspacePath } from '../../util';
 
+export enum CommandVersion {
```

Comment:
@mohanraj-r - What happens when we clean the legacy code in a couple of months? Do we still leave the command version when there is only one version of the commands?

Reply:
- As long as users are using older versions of CLI (which we have no control over) wouldn't we require legacy support?
- Regardless we can use this if this we ever have to do this legacy/beta switchover with any of the other cmds as well

Reply:
As long as users use older versions of CLI and newer version of extension, things would continue to work (no change in their workflow). It is when someone upgrades to new CLI and continue to use their old scratch org, the normal commands will not work and they'll have to use the legacy commands. This would continue only until the scratch orgs expire (1 to 30 days). Is the case you talking about when a user does not upgrade at all?

Got it. I do like the overall design and this is a nice legacy/beta switchover that can be extended to other commands as well in the future.

Reply:
Yes, thanks for the clarification Jag.